lambdas/postgres-migration-count-tool/src/index.ts

```diff
@@ -0,0 +1,203 @@
+* @param {number} [event.cutoffSeconds = 3600] - Number of seconds in the past
+* to count backward from. This allows you to explicitly set a window relative to
+* the execution of this tool to avoid 'leading edge' ingest processes that may
+* not have completed/populated elasticsearch/etc.
+* @returns {Promise<reportObj>} -- Returns a reportObj containing the report
+* outputs from the run
+*/
+export const handler = async (
+  event: {
+    dbConcurrency?: number,
+    dbMaxPool?: number,
+    reportBucket?: string,
+    reportPath?: string,
+    cutoffSeconds?: number,
+    systemBucket?: string,
+    stackName?: string,
+    // Arguments below are for unit test injection
+    getPostgresModelCountFunction?: typeof getPostgresModelCount,
+    mapperFunction?: typeof pMapMapper,
+    buildCollectionMappingsFunction?: typeof buildCollectionMappings,
+    getDynamoTableEntriesFunction?: typeof getDynamoTableEntries,
+    getKnexClientFunction?: typeof getKnexClient,
+  }
+): Promise<any> => {
+  const {
+    dbConcurrency = 20,
+    dbMaxPool = 20,
+    reportBucket,
+    reportPath,
+    cutoffSeconds = 3600,
+    systemBucket = envUtils.getRequiredEnvVar('systemBucket'),
+    stackName = envUtils.getRequiredEnvVar('prefix'),
+    getKnexClientFunction = getKnexClient,
+    getPostgresModelCountFunction = getPostgresModelCount,
+    mapperFunction = pMapMapper,
+    buildCollectionMappingsFunction = buildCollectionMappings,
+    getDynamoTableEntriesFunction = getDynamoTableEntries,
+  } = event;
+  process.env.dbMaxPool = `${dbMaxPool}`;
+
+  logger.debug(`Running reconciliation with ${JSON.stringify(event)}`);
+  const prefix = process.env.prefix || '';
+  const knexClient = await getKnexClientFunction({ env: process.env });
+  const cutoffTime = Date.now() - cutoffSeconds * 1000;
+  const cutoffIsoString = new Date(cutoffTime).toISOString();
+
+  const dynamoCollectionModel = new models.Collection();
+  const dynamoProvidersModel = new models.Provider();
+  const dynamoRulesModel = new models.Rule();
+
+  const dynamoAsyncOperationsModel = new models.AsyncOperation({
+    stackName,
+    systemBucket,
+  });
+
+  const postgresAsyncOperationModel = new AsyncOperationPgModel();
+  const postgresCollectionModel = new CollectionPgModel();
+  const postgresProviderModel = new ProviderPgModel();
+  const postgresRulesModel = new RulePgModel();
+
+  const [
+    dynamoCollections,
+    dynamoProviders,
+    dynamoRules,
+    dynamoAsyncOperations,
+  ] = await getDynamoTableEntriesFunction({
+    dynamoCollectionModel,
+    dynamoProvidersModel,
+    dynamoRulesModel,
+    dynamoAsyncOperationsModel,
+  });
+  const dynamoAsyncOperationsCount = dynamoAsyncOperations.length;
+  const dynamoCollectionsCount = dynamoCollections.length;
+  const dynamoProvidersCount = dynamoProviders.length;
+  const dynamoRuleCount = dynamoRules.length;
+
+  // Get postgres table counts
+  const postgresProviderCount = await getPostgresModelCountFunction({
```

Comment:
As these are functions, can we rename this to a verb-based thing like `countProstgresProviders`?

Reply:
`const postgresProviderCount` isn't a function, it's a number resolved from the promise return of `getPostgresModelCountFunction`, which is actually `getPostgresModelCount` (test injection) 

```
export const getPostgresModelCount = async <T, R extends { cumulus_id: number }>(params: {
  model: BasePgModel<T, R>,
  knexClient: Knex,
  cutoffIsoString?: string,
  queryParams?: ([string, string, string]|[Partial<R>])[],
}): Promise<number> => {
```

Reply:
But you're referring to `getPostgresModelCountFunction`...

Reply:
Also: 27cfe1e11

Reply:
27cfe1e11